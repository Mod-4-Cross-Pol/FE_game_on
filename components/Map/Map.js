import React from 'react';
import MapView, { Marker } from 'react-native-maps';
import { StyleSheet, Text, View } from 'react-native';

export default function Map({ navigation, events }) {
    return (
      <MapView
      showsUserLocation={true}
      style={styles.map}
      region={{
        latitude: 39.7306645,
        longitude: -104.988245,
        latitudeDelta: 0.1,
        longitudeDelta: 0.1
      }}
      >
      {events.map(event => {
        
        let eventLat = event.lat_long.split(',')[0];
        let eventLong = event.lat_long.split(',')[1];
        
        return <MapView.Marker
        coordinate=
        {{latitude: eventLat,
          longitude: eventLong}}
          title={event.activity}
          description={event.notes}
        onPress={() => {navigation.navigate('FullEventCard', {event: event})}}
        />
      })}
      </MapView>
    );
  }

const styles = StyleSheet.create({
  map: {
    height: 350,
    width: 'auto',
    borderWidth: 5,
    borderColor: '#0060b4'
  }
});
