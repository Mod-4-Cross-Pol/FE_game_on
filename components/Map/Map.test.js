import React from 'react';
import { Map } from './Map';
import renderer from 'react-test-renderer';

describe('Map', () => {

  describe('Renders a Map component', () => {

    // test('renders correctly', () => {
    //   const wrapper = renderer.create(<Map />).toJSON();
    //   expect(wrapper).toMatchSnapshot();
    // });

    test('it works', () => {
      expect(true).toBeTruthy();
    });

  });

});
