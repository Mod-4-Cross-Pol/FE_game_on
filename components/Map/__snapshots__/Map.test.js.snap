// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Renders a Map component renders the component 1`] = `ShallowWrapper {}`;
